src/nvim/lua/vim.lua

```diff
@@ -22,7 +22,7 @@ local function _update_package_paths()
       -- Find out path patterns. pathtrail should contain something like
       -- /?.so, /?/init.lua, /?.lua. This allows not to bother determining what
       -- correct suffixes are.
-      local pathtrail = s:match('[/\\][^/\\]*%?.*$')
+      local pathtrail = s:match('[/\\:][^/\\:]*%?.*$')
```

Comment:
Wouldn't it be more accurate to build a character class from `path.config:sub(1.1)`?